src/mappings/helpers/helpers.ts

```diff
@@ -916,6 +961,68 @@ export function updateAdvancedIndexerMetrics(indexer: Indexer): Indexer {
   return indexer as Indexer
 }
 
+export function calculateOwnStakeRatioForProvision(provision: Provision): BigDecimal {
+  let totalTokens = provision.tokensProvisioned.plus(provision.delegatedTokens)
+  return totalTokens == BigInt.fromI32(0)
+    ? BigDecimal.fromString('0')
+    : provision.tokensProvisioned.toBigDecimal().div(totalTokens.toBigDecimal())
+}
+
+export function calculateDelegatedStakeRatioForProvision(provision: Provision): BigDecimal {
+  let totalTokens = provision.tokensProvisioned.plus(provision.delegatedTokens)
+  return totalTokens == BigInt.fromI32(0)
+    ? BigDecimal.fromString('0')
+    : provision.delegatedTokens.toBigDecimal().div(totalTokens.toBigDecimal())
+}
+
+export function calculateIndexingRewardEffectiveCutForProvision(provision: Provision): BigDecimal {
+  let delegatorCut =
+    provision.indexingRewardsCut.toBigDecimal() /
+    BigDecimal.fromString('1000000')
+  return provision.delegatedStakeRatio == BigDecimal.fromString('0')
+    ? BigDecimal.fromString('0')
+    : BigDecimal.fromString('1') - delegatorCut / provision.delegatedStakeRatio
+}
+
+export function calculateQueryFeeEffectiveCutForProvision(provision: Provision): BigDecimal {
+  let delegatorCut =
+    provision.queryFeeCut.toBigDecimal() / BigDecimal.fromString('1000000')
+  return provision.delegatedStakeRatio == BigDecimal.fromString('0')
+    ? BigDecimal.fromString('0')
+    : BigDecimal.fromString('1') - delegatorCut / provision.delegatedStakeRatio
+}
+
+export function calculateIndexerRewardOwnGenerationRatioForProvision(provision: Provision): BigDecimal {
+  let rewardCut =
+    provision.indexingRewardsCut.toBigDecimal() / BigDecimal.fromString('1000000')
+  return provision.ownStakeRatio == BigDecimal.fromString('0')
+    ? BigDecimal.fromString('0')
+    : rewardCut / provision.ownStakeRatio
```

Comment:
```suggestion
    : (BigDecimal.fromString('1') - rewardCut) / provision.ownStakeRatio
```

I think this is the correct calculation. What we are looking for (as far as I understood from the schema definition), is the ratio between the `total query fees obtained by the indexer` and the `total query fees generated by the indexer's stake`. A ratio > 1 indicates the indexer is getting some fees from the delegator's stake, a ratio < 1 the opposite. 

Is this correct?

Reply:
yeah, I think I missed updating this one, since all previous ones were updated (and changed the naming from `rewardCut` to `delegatorCut`, but this one wasn't 😅